pthread/pthread.c

```diff
@@ -594,3 +665,131 @@ int pthread_cond_timedwait(pthread_cond_t *restrict cond,
 	}
 	return err;
 }
+
+
+int pthread_sigmask(int how, const sigset_t *__restrict__ set, sigset_t *__restrict__ oset)
+{
+	int ret = sigprocmask(how, set, oset);
+	return (ret == -1) ? -errno : EOK;
+}
+
+
+int pthread_kill(pthread_t thread, int sig)
+{
+	int ret = kill(thread, sig);
+	return (ret == -1) ? -errno : EOK;
+}
+
+
+int pthread_key_create(pthread_key_t *key, void (*destructor)(void *))
+{
+	int err = EOK;
+	*key = malloc(sizeof(__pthread_key_t));
+	if (key == NULL) {
+		err = -ENOMEM;
+	}
+	else {
+		(*key)->destructor = destructor;
+	}
+	return err;
+}
+
+
+int pthread_key_delete(pthread_key_t key)
+{
+	int err = EOK;
```

Comment:
It seems to me that we never modify err, it always stays EOK. Maybe we should return error if key hasn't been found (and skip free(key) if so)?

Reply:
I don't think there is a way to determine if key is valid or not, and whether it should be freed. Someone could call `pthread_key_create` - which mallocs memory for` __pthread_key_t` - and then not use this key at all but when `pthread_key_delete` is called the memory still has to be freed. I'll just return `EOK`.